allocator.go

```diff
@@ -106,6 +115,18 @@ func WithTagMatching() Option {
 	}
 }
 
+func WithChurnConstraint(prevAssignment map[RangeID][]NodeID, minimizeToggle bool, maxChurn int64) Option {
+	return func(opt *options) {
+		opt.churnOptions.prevAssignment = prevAssignment
+		opt.churnOptions.WithMinimalChurn = minimizeToggle
+		opt.churnOptions.withMaxChurn = maxChurn < math.MaxInt64
+		if maxChurn < 0 {
+			panic("max churn should be more than 0")
```

Comment:
Go's default error messages don't start with uppercase or end with period. I guess following that would be appropriate but I am not sure.

Reply:
Reading Go defined panic messages and that indeed seems to be the case. Maybe it's a ploy to convince developers of the error's severity, i.e whatever you did got me[Go] panicking so bad that I forgot capitalization & punctuation. 😛

Reply:
The lowercase is indeed the convention; it's because sometimes error messages can be combined upper in the stack and using capitals mid sentence ends up not composing well.